src/main/java/io/github/lottetreg/httpserver/ServerSocket.java

```diff
@@ -0,0 +1,26 @@
+package io.github.lottetreg.httpserver;
+
+import java.net.Socket;
+
+public class ServerSocket {
+  private java.net.ServerSocket serverSocket;
+
+  ServerSocket(java.net.ServerSocket serverSocket) {
+    this.serverSocket = serverSocket;
+  }
+
+  public Connection acceptConnection() {
+    try {
+      Socket socket = this.serverSocket.accept();
+      return new Connection(socket);
+    } catch (Exception e) {
+      throw new FailedToAcceptConnectionException(e);
+    }
+  }
+
+  static class FailedToAcceptConnectionException extends RuntimeException {
```

Comment:
I like that you've created your own custom exceptions, is there a reason you decided to make them static classes?

Reply:
Thank you for pointing this out, Kara! I originally just did this in the `Connection` class (and mistakenly did it everywhere else after), because the `WriterTest` tests were failing with this error: "an enclosing instance that contains Connection.FailedToGetInputStreamException is required", which I understand to mean that I can't create a new `Connection.FailedToGetInputStreamException` without it belonging to an instance of `Connection`. I've just moved the `Connection` Exception classes into the `Connectionable` interface, which fixes the problem. I'm not sure if it's best practice, but the Exception classes don't feel out of place in there...